gcs-action/src/gcs.ts

```diff
@@ -0,0 +1,202 @@
+import {
+  Storage,
+  UploadResponse,
+  DeleteFileResponse,
+  File
+} from '@google-cloud/storage'
+import * as fs from 'fs'
+import * as path from 'path'
+import * as core from '@actions/core'
+
+const storage = new Storage()
```

Comment:
This assumes ADC have been setup in the env. What if a user wants to use this action independently of the setup-gcloud action? I think we should support both use-cases by adding support for specifying an SA key as an optional action param.

Reply:
Added support for SA key as optional param